[Bug]  java.lang.ClassCastException: class org.apache.hugegraph.task.DistributedTaskScheduler cannot be cast to class org.apache.hugegraph.task.StandardTaskScheduler

### Bug Type (问题类型)

None

### Before submit

- [x] 我已经确认现有的 [Issues](https://github.com/apache/hugegraph/issues) 与 [FAQ](https://hugegraph.apache.org/docs/guides/faq/) 中没有相同 / 重复问题 (I have confirmed and searched that there are no similar problems in the historical issue and documents)

### Environment (环境信息)

- Server Version: 1.5.0 (Apache Release Version)
- Backend: Hstore x nodes, HDD or SSD 
- OS: xx CPUs, xx G RAM, Ubuntu 2x.x / CentOS 7.x 
- Data Size:  xx vertices, xx edges <!-- (like 1000W 点, 9000W 边) -->


### Expected & Actual behavior (期望与实际表现)

2025-07-28 17:56:05 [pool-15-thread-1] [ERROR] o.a.h.t.TaskManager - Exception occurred when change to master role
java.lang.ClassCastException: class org.apache.hugegraph.task.DistributedTaskScheduler cannot be cast to class org.apache.hugegraph.task.StandardTaskScheduler (org.apache.hugegraph.task.DistributedTaskScheduler and org.apache.hugegraph.task.StandardTaskScheduler are in unnamed module of loader 'app')
        at org.apache.hugegraph.task.TaskManager.onAsRoleMaster(TaskManager.java:359) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at org.apache.hugegraph.masterelection.StandardRoleListener.onAsRoleMaster(StandardRoleListener.java:47) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at org.apache.hugegraph.masterelection.StandardRoleElectionStateMachine.apply(StandardRoleElectionStateMachine.java:75) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at org.apache.hugegraph.masterelection.StandardRoleElectionStateMachine.lambda$start$0(StandardRoleElectionStateMachine.java:60) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]
2025-07-28 17:56:05 [pool-15-thread-1] [ERROR] o.a.h.m.StandardRoleListener - Server server-1 exception occurred
java.lang.ClassCastException: class org.apache.hugegraph.task.DistributedTaskScheduler cannot be cast to class org.apache.hugegraph.task.StandardTaskScheduler (org.apache.hugegraph.task.DistributedTaskScheduler and org.apache.hugegraph.task.StandardTaskScheduler are in unnamed module of loader 'app')
        at org.apache.hugegraph.task.TaskManager.onAsRoleMaster(TaskManager.java:359) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at org.apache.hugegraph.masterelection.StandardRoleListener.onAsRoleMaster(StandardRoleListener.java:47) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at org.apache.hugegraph.masterelection.StandardRoleElectionStateMachine.apply(StandardRoleElectionStateMachine.java:75) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at org.apache.hugegraph.masterelection.StandardRoleElectionStateMachine.lambda$start$0(StandardRoleElectionStateMachine.java:60) ~[hugegraph-core-1.5.0.jar:1.5.0]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at java.lang.Thread.run(Thread.java:829) [?:?]

### Vertex/Edge example (问题点 / 边数据举例)

```javascript
我的rest配置

restserver.url=http://0.0.0.0:8081
gremlinserver.url=http://127.0.0.1:8181

graphs=./conf/graphs

# ????
auth.authenticator=org.apache.hugegraph.auth.StandardAuthenticator
auth.admin_token=54702de7-b89e-4a80-817c-b3020e7323a0
auth.user_tokens=user0:d75469f7-90e8-45fd-9012-7801cc0d9df6

rpc.server_host=127.0.0.1
rpc.server_port=8091

rpc.remote_url=127.0.0.1:8091
rpc.client_connect_timeout=60
rpc.client_reconnect_period=15
rpc.client_read_timeout=60
rpc.client_retries=5
rpc.client_load_balancer=consistentHash


log.slow_query_threshold=1000

server.id=server-1
server.role=master
```

### Schema [VertexLabel, EdgeLabel, IndexLabel] (元数据结构)

```javascript

```

--- a/hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/config/ServerOptions.java
+++ b/hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/config/ServerOptions.java
@@ -73,6 +73,15 @@ public class ServerOptions extends OptionHolder {
                     "master"
             );
 
+    public static final ConfigOption<Boolean> ENABLE_SERVER_ROLE_ELECTION =
+            new ConfigOption<>(
+                    "server.role_election",
+                    "Whether to enable role election, if enabled, the server " +
+                    "will elect a master node in the cluster.",
+                    disallowEmpty(),
+                    true
+            );
+
     public static final ConfigOption<Integer> MAX_WORKER_THREADS =
             new ConfigOption<>(
                     "restserver.max_worker_threads",

--- a/hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/config/ServerOptions.java
+++ b/hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/config/ServerOptions.java
@@ -79,7 +79,7 @@ public class ServerOptions extends OptionHolder {
                     "Whether to enable role election, if enabled, the server " +
                     "will elect a master node in the cluster.",
                     disallowEmpty(),
-                    true
+                    false
             );
 
     public static final ConfigOption<Integer> MAX_WORKER_THREADS =

--- a/hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/core/GraphManager.java
+++ b/hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/core/GraphManager.java
@@ -469,7 +469,8 @@ public final class GraphManager {
 
         NodeRole nodeRole = NodeRole.valueOf(role.toUpperCase());
         boolean supportRoleElection = !nodeRole.computer() &&
-                                      this.supportRoleElection();
+                                      this.supportRoleElection() &&
+                                      config.get(ServerOptions.ENABLE_SERVER_ROLE_ELECTION);
         if (supportRoleElection) {
             // Init any server as Worker role, then do role election
             nodeRole = NodeRole.WORKER;

--- a/hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/task/TaskManager.java
+++ b/hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/task/TaskManager.java
@@ -37,6 +37,13 @@ import org.apache.hugegraph.util.LockUtil;
 import org.apache.hugegraph.util.Log;
 import org.slf4j.Logger;
 
+/**
+ * Central task management system that coordinates task scheduling and execution.
+ * Manages task schedulers for different graphs and handles role-based execution.
+ * <p>
+ * Note: The local master-worker mechanism will be deprecated in version 1.7
+ * (configuration has been removed from config files).
+ */
 public final class TaskManager {
 
     private static final Logger LOG = Log.logger(TaskManager.class);
@@ -44,8 +51,7 @@ public final class TaskManager {
     public static final String TASK_WORKER_PREFIX = "task-worker";
     public static final String TASK_WORKER = TASK_WORKER_PREFIX + "-%d";
     public static final String TASK_DB_WORKER = "task-db-worker-%d";
-    public static final String SERVER_INFO_DB_WORKER =
-            "server-info-db-worker-%d";
+    public static final String SERVER_INFO_DB_WORKER = "server-info-db-worker-%d";
     public static final String TASK_SCHEDULER = "task-scheduler-%d";
 
     public static final String OLAP_TASK_WORKER = "olap-task-worker-%d";
@@ -53,7 +59,7 @@ public final class TaskManager {
     public static final String EPHEMERAL_TASK_WORKER = "ephemeral-task-worker-%d";
     public static final String DISTRIBUTED_TASK_SCHEDULER = "distributed-scheduler-%d";
 
-    protected static final long SCHEDULE_PERIOD = 1000L; // unit ms
+    static final long SCHEDULE_PERIOD = 1000L; // unit ms
     private static final long TX_CLOSE_TIMEOUT = 30L; // unit s
     private static final int THREADS = 4;
     private static final TaskManager MANAGER = new TaskManager(THREADS);
@@ -87,17 +93,13 @@ public final class TaskManager {
         this.serverInfoDbExecutor = ExecutorUtil.newFixedThreadPool(
                 1, SERVER_INFO_DB_WORKER);
 
-        this.schemaTaskExecutor = ExecutorUtil.newFixedThreadPool(pool,
-                                                                  SCHEMA_TASK_WORKER);
-        this.olapTaskExecutor = ExecutorUtil.newFixedThreadPool(pool,
-                                                                OLAP_TASK_WORKER);
-        this.ephemeralTaskExecutor = ExecutorUtil.newFixedThreadPool(pool,
-                                                                     EPHEMERAL_TASK_WORKER);
+        this.schemaTaskExecutor = ExecutorUtil.newFixedThreadPool(pool, SCHEMA_TASK_WORKER);
+        this.olapTaskExecutor = ExecutorUtil.newFixedThreadPool(pool, OLAP_TASK_WORKER);
+        this.ephemeralTaskExecutor = ExecutorUtil.newFixedThreadPool(pool, EPHEMERAL_TASK_WORKER);
         this.distributedSchedulerExecutor =
-                ExecutorUtil.newPausableScheduledThreadPool(1,
-                                                            DISTRIBUTED_TASK_SCHEDULER);
+                ExecutorUtil.newPausableScheduledThreadPool(1, DISTRIBUTED_TASK_SCHEDULER);
 
-        // For schedule task to run, just one thread is ok
+        // For a schedule task to run, just one thread is ok
         this.schedulerExecutor = ExecutorUtil.newPausableScheduledThreadPool(
                 1, TASK_SCHEDULER);
         // Start after 10x period time waiting for HugeGraphServer startup
@@ -111,7 +113,9 @@ public final class TaskManager {
         E.checkArgumentNotNull(graph, "The graph can't be null");
         LOG.info("Use {} as the scheduler of graph ({})",
                  graph.schedulerType(), graph.name());
-        // TODO: If the current service is bound to a specified non-DEFAULT graph space, the graph outside of the current graph space will no longer create task schedulers (graph space)
+        // TODO: If the current service is bound to a specified non-DEFAULT graph space, the
+        //  graph outside of the current graph space will no longer create task schedulers (graph
+        //  space)
         switch (graph.schedulerType()) {
             case "distributed": {
                 TaskScheduler scheduler =
@@ -194,7 +198,7 @@ public final class TaskManager {
 
     private void closeSchedulerTx(HugeGraphParams graph) {
         final Callable<Void> closeTx = () -> {
-            // Do close-tx for current thread
+            // Do close-tx for the current thread
             graph.closeTx();
             // Let other threads run
             Thread.yield();
@@ -209,7 +213,7 @@ public final class TaskManager {
 
     private void closeDistributedSchedulerTx(HugeGraphParams graph) {
         final Callable<Void> closeTx = () -> {
-            // Do close-tx for current thread
+            // Do close-tx for the current thread
             graph.closeTx();
             // Let other threads run
             Thread.yield();
@@ -252,8 +256,7 @@ public final class TaskManager {
         if (!this.schedulerExecutor.isShutdown()) {
             this.schedulerExecutor.shutdown();
             try {
-                terminated = this.schedulerExecutor.awaitTermination(timeout,
-                                                                     unit);
+                terminated = this.schedulerExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -262,8 +265,7 @@ public final class TaskManager {
         if (terminated && !this.distributedSchedulerExecutor.isShutdown()) {
             this.distributedSchedulerExecutor.shutdown();
             try {
-                terminated = this.distributedSchedulerExecutor.awaitTermination(timeout,
-                                                                                unit);
+                terminated = this.distributedSchedulerExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -272,8 +274,7 @@ public final class TaskManager {
         if (terminated && !this.taskExecutor.isShutdown()) {
             this.taskExecutor.shutdown();
             try {
-                terminated = this.taskExecutor.awaitTermination(timeout,
-                                                                unit);
+                terminated = this.taskExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -282,8 +283,7 @@ public final class TaskManager {
         if (terminated && !this.serverInfoDbExecutor.isShutdown()) {
             this.serverInfoDbExecutor.shutdown();
             try {
-                terminated = this.serverInfoDbExecutor.awaitTermination(timeout,
-                                                                        unit);
+                terminated = this.serverInfoDbExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -292,8 +292,7 @@ public final class TaskManager {
         if (terminated && !this.taskDbExecutor.isShutdown()) {
             this.taskDbExecutor.shutdown();
             try {
-                terminated = this.taskDbExecutor.awaitTermination(timeout,
-                                                                  unit);
+                terminated = this.taskDbExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -302,8 +301,7 @@ public final class TaskManager {
         if (terminated && !this.ephemeralTaskExecutor.isShutdown()) {
             this.ephemeralTaskExecutor.shutdown();
             try {
-                terminated = this.ephemeralTaskExecutor.awaitTermination(timeout,
-                                                                         unit);
+                terminated = this.ephemeralTaskExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -312,8 +310,7 @@ public final class TaskManager {
         if (terminated && !this.schemaTaskExecutor.isShutdown()) {
             this.schemaTaskExecutor.shutdown();
             try {
-                terminated = this.schemaTaskExecutor.awaitTermination(timeout,
-                                                                      unit);
+                terminated = this.schemaTaskExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -322,8 +319,7 @@ public final class TaskManager {
         if (terminated && !this.olapTaskExecutor.isShutdown()) {
             this.olapTaskExecutor.shutdown();
             try {
-                terminated = this.olapTaskExecutor.awaitTermination(timeout,
-                                                                    unit);
+                terminated = this.olapTaskExecutor.awaitTermination(timeout, unit);
             } catch (Throwable e) {
                 ex = e;
             }
@@ -360,8 +356,7 @@ public final class TaskManager {
                 if (serverInfoManager != null) {
                     serverInfoManager.changeServerRole(NodeRole.MASTER);
                 } else {
-                    LOG.warn("ServerInfoManager is null for graph {}",
-                             entry.graphName());
+                    LOG.warn("ServerInfoManager is null for graph {}", entry.graphName());
                 }
             }
         } catch (Throwable e) {
@@ -377,8 +372,7 @@ public final class TaskManager {
                 if (serverInfoManager != null) {
                     serverInfoManager.changeServerRole(NodeRole.WORKER);
                 } else {
-                    LOG.warn("ServerInfoManager is null for graph {}",
-                             entry.graphName());
+                    LOG.warn("ServerInfoManager is null for graph {}", entry.graphName());
                 }
             }
         } catch (Throwable e) {
@@ -387,8 +381,8 @@ public final class TaskManager {
         }
     }
 
-    protected void notifyNewTask(HugeTask<?> task) {
-        Queue<Runnable> queue = ((ThreadPoolExecutor) this.schedulerExecutor)
+    void notifyNewTask(HugeTask<?> task) {
+        Queue<Runnable> queue = this.schedulerExecutor
                 .getQueue();
         if (queue.size() <= 1) {
             /*
@@ -406,10 +400,9 @@ public final class TaskManager {
         // Called by scheduler timer
         try {
             for (TaskScheduler entry : this.schedulers.values()) {
-                TaskScheduler scheduler = entry;
-                // Maybe other thread close&remove scheduler at the same time
-                synchronized (scheduler) {
-                    this.scheduleOrExecuteJobForGraph(scheduler);
+                // Maybe other threads close&remove scheduler at the same time
+                synchronized (entry) {
+                    this.scheduleOrExecuteJobForGraph(entry);
                 }
             }
         } catch (Throwable e) {

--- a/hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/task/TaskManager.java
+++ b/hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/task/TaskManager.java
@@ -356,9 +356,13 @@ public final class TaskManager {
     public void onAsRoleMaster() {
         try {
             for (TaskScheduler entry : this.schedulers.values()) {
-                StandardTaskScheduler scheduler = (StandardTaskScheduler) entry;
-                ServerInfoManager serverInfoManager = scheduler.serverManager();
-                serverInfoManager.changeServerRole(NodeRole.MASTER);
+                ServerInfoManager serverInfoManager = entry.serverManager();
+                if (serverInfoManager != null) {
+                    serverInfoManager.changeServerRole(NodeRole.MASTER);
+                } else {
+                    LOG.warn("ServerInfoManager is null for graph {}",
+                             entry.graphName());
+                }
             }
         } catch (Throwable e) {
             LOG.error("Exception occurred when change to master role", e);
@@ -369,9 +373,13 @@ public final class TaskManager {
     public void onAsRoleWorker() {
         try {
             for (TaskScheduler entry : this.schedulers.values()) {
-                StandardTaskScheduler scheduler = (StandardTaskScheduler) entry;
-                ServerInfoManager serverInfoManager = scheduler.serverManager();
-                serverInfoManager.changeServerRole(NodeRole.WORKER);
+                ServerInfoManager serverInfoManager = entry.serverManager();
+                if (serverInfoManager != null) {
+                    serverInfoManager.changeServerRole(NodeRole.WORKER);
+                } else {
+                    LOG.warn("ServerInfoManager is null for graph {}",
+                             entry.graphName());
+                }
             }
         } catch (Throwable e) {
             LOG.error("Exception occurred when change to worker role", e);

